pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -1142,7 +1143,12 @@ func serviceUpdateNeedsPush(prev, curr *model.Service) bool {
 		return true
 	}
 	if prev == nil {
-		return true
+		return !curr.Attributes.ExportTo.Contains(visibility.None)
+	}
+	// if service are not exported, no need to push
+	if prev.Attributes.ExportTo.Contains(visibility.None) &&
```

Comment:
Should we not cache service when it is not visible at all

Reply:
Updated. Invisible services will be regarded as non-existent.